src/chapter08-porting-wolfHSM.md

```diff
@@ -1 +1,153 @@
+
+- Standalone Reference Server Application: This application is meant to run on the HSM core and handle all secure operations. It comes fully functional out-of-the-box but can also be customized by the end user to support additional use cases
+- Client Library: This library can be linked against user applications to facilitate communication with the server
+
+# Ports
+
+## Infineon Aurix TC3XX
+
+(Port in progress)
+The distribution of this  port is restricted by the vendor.  Please contact support@wolfssl.com for access. 
+
+Infineon Aurix TC3xx
+- Up to 6x 300MHz TriCore application cores
+- 1x 100MHz ARM Cortex M3 HSM core
+- Crypto offload: TRNG, AES128, ECDSA, ED25519, SHA
+
+## ST SPC58NN
+
+(Port in progress)
+The distribution of this  port is restricted by the vendor.  Please contact support@wolfssl.com for access. 
+
+ST SPC58NN
+- 3x 200MHz e200z4256 PowerPC application cores
+- 1x 100MHz e200z0 PowerPC HSM core with NVM
+- Crypto offload: TRNG, AES128
+
+## Posix
+
+The Posix port provides multiple and fully functional implementations of different wolfHSM abstractions that can be used to better understand the exact functionality expected for different hardware abstractions.
+
+The Posix port provides:
+- Memory buffer transport
+- TCP transport
+- Unix domain transport
+- NVM device (using a filesystem)
+- Flash device (using a file as a backing store)
+
+## Skeleton 
+
+The Skeleton port source code provides a non-functioning layout to be used as a starting point for future hardware/platform ports.  Each function provides the basic description and expected flow with error cases explained so that ports can be used interchangeably with consistent results.
+
+The Skeleton  port provides implementations of:
+- Transport 
+- NVM Device
+- Flash Device
+- Crytp Device
+
+# WolfHSM Porting Interface
+
+## WolfHSM Porting Interface Explained 
+When porting WolfHSM into a new platform or ours, you will need to implement some hardware-specific interfaces. We can provide helper functions for flash-like devices. You will need to provide (read, erase, program, blank check, etc.). Our library will likely work on top of it. We can then provide the object layer on top of that. You will also need to provide crypto hardware interfaces like TRNG, Keys, and symmetric/asymmetric crypto. From a platform interface perspective, you will have to understand your system's boot sequence and how the application cores get reset or marked to boot off of it at a specific start location. You may also need to control shared memory locations and memory translation units. The configuration you want to select is selected at compile time rather than link time. In your code, you select which libraries you want to use; here's the config of the libraries, and then work that at compile time, which is an AUTOSAR requirement. 
```

Comment:
The wording of this paragraph is a little confusing, and is a little redundant, I might say just omit it